UI/basicGUI/src/pages/home/home.ts

```diff
@@ -13,19 +13,52 @@ export class HomePage {
   map: any;
   inputLocation = '';
   inputDestination = '';
-  constructor(public navCtrl: NavController, public alertCtrl: AlertController) {
+  currentLatlng: any;
+  marker: leaflet.market
 
+  constructor(public navCtrl: NavController, public alertCtrl: AlertController) {
   }
 
   ionViewDidEnter() {
     this.loadmap();
   }
 
+   getLocation(){
+     this.map.locate({
+     }).once('locationfound', (e) => {
+      this.currentLatlng = e.latlng
+
+      let markerGroup = leaflet.featureGroup();
+      if (this.marker) { // check
```

Comment:
should you remove comments on line 32 and 33?
"// check"

Reply:
i explained the code there with a new comment, let me know if i should delete it